### ISSUE TITLE
Search for commit by sha

### ISSUE THREAD

**Issue by thedillonb** [thedillonb/CodeHub#204]

**Comment by steakknife** [comment]
Yes, I was looking exactly for this. It should take both partial and full hashes. Maybe the search bar in the Commits view could be multiplexed so it dynamically finds both text and hash matches?

It would be awesome brownie points if the search results showed a small, short hash in the right top corner and a copy icon that copies the full hash to the clipboard too. 📋
